tests/test_compile.py

```diff
@@ -0,0 +1,87 @@
+"""
+Test protoc code generator plugin for custom protoc message types
+"""
+import unittest
+import tempfile
+import subprocess
+import os
+
+# Import into globals() so generated code string can be compiled
+from snowflake.telemetry._internal.serialize import ProtoSerializer, Enum
+
+class TestCompile(unittest.TestCase):
+    def test_compile(self):
+        with tempfile.NamedTemporaryFile(suffix=".proto", mode="w", delete=False) as proto_file:
+            # Define a simple proto file
+            proto_file.write("""syntax = "proto3";
+package opentelemetry.proto.common.v1;
+
+message AnyValue {
+  oneof value {
+    string string_value = 1;
+    bool bool_value = 2;
```

Comment:
Would it be easier if this is a seperate file in, say, `tests/data/simple.proto`